src/librustc_codegen_ssa/back/write.rs

```diff
@@ -0,0 +1,1826 @@
+    time_graph::TimelineId(CODEGEN_WORKER_ID);
+pub const CODEGEN_WORK_PACKAGE_KIND: time_graph::WorkPackageKind =
+    time_graph::WorkPackageKind(&["#DE9597", "#FED1D3", "#FDC5C7", "#B46668", "#88494B"]);
+const LLVM_WORK_PACKAGE_KIND: time_graph::WorkPackageKind =
+    time_graph::WorkPackageKind(&["#7DB67A", "#C6EEC4", "#ACDAAA", "#579354", "#3E6F3C"]);
+
+// Set up a destructor which will fire off a message that we're done as
+// we exit.
+struct Bomb<B : ExtraBackendMethods> {
+    coordinator_send: Sender<Box<dyn Any + Send>>,
+    result: Option<WorkItemResult<B::Module>>,
+    worker_id: usize,
+}
+impl<B : ExtraBackendMethods> Drop for Bomb<B> {
+    fn drop(&mut self) {
+        let worker_id = self.worker_id;
+        let msg : Message<B> = match self.result.take() {
+            Some(WorkItemResult::Compiled(m)) => {
+                Message::Done { result: Ok(m), worker_id }
+            }
+            Some(WorkItemResult::NeedsLTO(m)) => {
+                Message::NeedsLTO { result: m, worker_id }
+            }
+            None => Message::Done { result: Err(()), worker_id }
+        };
+        drop(self.coordinator_send.send(Box::new(msg)));
+    }
+}
+
+fn spawn_work<B : ExtraBackendMethods>(
+    cgcx: CodegenContext<B>,
+    work: WorkItem<B>
+) {
+    let depth = time_depth();
+
+    thread::spawn(move || {
+        set_time_depth(depth);
+
+        let mut bomb : Bomb<B> = Bomb {
+            coordinator_send: cgcx.coordinator_send.clone(),
+            result: None,
+            worker_id: cgcx.worker,
+        };
+
+        // Execute the work itself, and if it finishes successfully then flag
+        // ourselves as a success as well.
+        //
+        // Note that we ignore any `FatalError` coming out of `execute_work_item`,
+        // as a diagnostic was already sent off to the main thread - just
+        // surface that there was an error in this worker.
+        bomb.result = {
+            let timeline = cgcx.time_graph.as_ref().map(|tg| {
+                tg.start(time_graph::TimelineId(cgcx.worker),
+                         LLVM_WORK_PACKAGE_KIND,
+                         &work.name())
+            });
+            let mut timeline = timeline.unwrap_or(Timeline::noop());
+            execute_work_item(&cgcx, work, &mut timeline).ok()
+        };
+    });
+}
+
+pub fn run_assembler<B : ExtraBackendMethods>(
+    cgcx: &CodegenContext<B>,
+    handler: &Handler,
+    assembly: &Path,
+    object: &Path
+) {
+    let assembler = cgcx.assembler_cmd
+        .as_ref()
+        .expect("cgcx.assembler_cmd is missing?");
+
+    let pname = &assembler.name;
+    let mut cmd = assembler.cmd.clone();
+    cmd.arg("-c").arg("-o").arg(object).arg(assembly);
+    debug!("{:?}", cmd);
+
+    match cmd.output() {
+        Ok(prog) => {
+            if !prog.status.success() {
+                let mut note = prog.stderr.clone();
+                note.extend_from_slice(&prog.stdout);
+
+   
```

Comment:
Maybe someone should search for `llvm` (case-insensitive) in `rustc_codegen_ssa`.